ciphers/hill_cipher.py

```diff
@@ -84,8 +105,13 @@ def check_determinant(self):
             )
 
     def process_text(self, text):
-        text = list(text.upper())
-        text = [char for char in text if char in self.key_string]
+        """
+        >>> HillCipher(numpy.matrix([[2,5],[1,6]])).process_text('testing hill cipher')
+        'TESTINGHILLCIPHERR'
+        >>>
+        """
+        text: list = list(text.upper())
+        text: list = [char for char in text if char in self.key_string]
```

Comment:
It is too confusing to use the same variable as both a str and as a list.  Can we replace lines 113 and 114 with:
`chars = [char for char in text.upper() if char in self.key_string]`
and then change the variable name in the lines below?  We do not need to declare the type because CPython is smart enough to know that a list comprehension returns a list.